kolibri/core/tasks/scheduler.py

```diff
@@ -249,3 +250,23 @@ def _ns_query(self, session):
 
     def _now(self):
         return local_now()
+
+    def is_func_scheduled(self, func):
```

Comment:
I think there's a simpler solution here.

When we schedule a task: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/tasks/scheduler.py#L143 we are able to pass in arbitrary args and kwargs for the job constructor.

If we update the job constructor to allow a supplied job_id to be passed in:
https://github.com/learningequality/kolibri/blob/develop/kolibri/core/tasks/job.py#L90:
`self.job_id = kwargs.get("job_id", uuid.uuid4().hex)`

Then we can just choose a constant job_id for the scheduled pingback task and just check that it exists.

This also has the virtue that we can schedule additional tasks for the `_ping` if we want.

Reply:
Wow :heart:, your approach is so simple!  

I just have one nit thought, don't you think assigning constant job ids might become an issue when jobs like ping are in good numbers...?

Reply:
We should expect that the `schedule_ping` function just ensures that there is a single ping scheduled - if other `_ping` jobs need to be scheduled, then they can not pass the constant job_id, and so will be assigned a random uuid.